sui_core/src/consensus_client.rs

```diff
@@ -64,18 +65,22 @@ impl ConsensusClient {
         buffer_size: usize,
     ) -> JoinHandle<SuiResult<()>> {
         log::info!("Consensus client connecting to {}", address);
-        tokio::spawn(async move {
-            handler.synchronize().await?;
-            handler.run(address, buffer_size).await?;
-            Ok(())
-        })
+        tokio::spawn(async move { handler.run(address, buffer_size).await })
     }
 
     /// Synchronize with the consensus in case we missed part of its output sequence.
     /// It is safety-critical that we process the consensus outputs in the right order.
-    async fn synchronize(&mut self) -> SuiResult<()> {
-        // TODO [issue #932]: [liveness-critical] Implement the synchronizer.
-        Ok(())
+    async fn synchronize(&mut self, connection: &mut TcpDataStream) -> SuiResult<()> {
+        let request = ConsensusSync {
+            sequencer_number: self.last_consensus_index,
+        };
+        let bytes = serialize_consensus_sync(&request);
+        connection
+            .write_data(&bytes)
```

Comment:
My advice: use the send and next functions on the Stream and Sink interfaces for networking.

Reply:
fixed